From 497d4bb13ba5b9d3054a3457bc7ca838f7ba4202 Mon Sep 17 00:00:00 2001
From: Florian Westphal <fwestpha@redhat.com>
Date: Tue, 2 Sep 2014 10:15:10 -0400
Subject: [net] skbuff: Use ALIGN macro instead of open coding it

Message-id: <1409652920-431-48-git-send-email-fwestpha@redhat.com>
Patchwork-id: 90161
O-Subject: [PATCH 47/57] [RHEL7.1 net] net: skbuff: Use ALIGN macro instead of open coding it
Bugzilla: 1131999
RH-Acked-by: Jiri Pirko <jpirko@redhat.com>
RH-Acked-by: Vlad Yasevich <vyasevic@redhat.com>
RH-Acked-by: Nikolay Aleksandrov <nikolay@redhat.com>
RH-Acked-by: Ivan Vecera <ivecera@redhat.com>
RH-Acked-by: Jiri Benc <jbenc@redhat.com>

dependency to avoid mangling 77cffe23c1f
("net: Clarification of CHECKSUM_UNNECESSARY")

commit 0bec8c88dc2b076a0a4a0437e1e878026cbaccb4
Author: Tobias Klauser <tklauser@distanz.ch>
Date:   Tue Jul 22 12:06:23 2014 +0200

    net: skbuff: Use ALIGN macro instead of open coding it

    Use ALIGN from linux/kernel.h to define SKB_DATA_ALIGN instead of open
    coding it.

    Signed-off-by: Tobias Klauser <tklauser@distanz.ch>
    Signed-off-by: David S. Miller <davem@davemloft.net>

Signed-off-by: Jiri Benc <jbenc@redhat.com>
Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/include/linux/skbuff.h b/include/linux/skbuff.h
index 3518bd6..dc940c9 100644
--- a/include/linux/skbuff.h
+++ b/include/linux/skbuff.h
@@ -40,8 +40,7 @@
 #define CHECKSUM_COMPLETE 2
 #define CHECKSUM_PARTIAL 3
 
-#define SKB_DATA_ALIGN(X) (((X) + (SMP_CACHE_BYTES - 1)) & \
-     ~(SMP_CACHE_BYTES - 1))
+#define SKB_DATA_ALIGN(X) ALIGN(X, SMP_CACHE_BYTES)
 #define SKB_WITH_OVERHEAD(X) \
  ((X) - SKB_DATA_ALIGN(sizeof(struct skb_shared_info)))
 #define SKB_MAX_ORDER(X, ORDER) \
-- 
1.7.1